test/e2e/suites/unmanaged/unmanaged_functional_test.go

```diff
@@ -1070,6 +1071,42 @@ var _ = ginkgo.Context("[unmanaged] [functional]", func() {
 			awsCluster, err := GetAWSClusterByName(ctx, namespace.Name, clusterName)
 			Expect(err).To(BeNil())
 
+			ginkgo.By("Creating a MachineDeployment bootstrapped via Ignition with StorageType UnencryptedUserData")
+			unencryptedMDName := clusterName + "-md-unencrypted-userdata"
+			unencryptedUDMachineTemplate := makeAWSMachineTemplate(namespace.Name, unencryptedMDName, e2eCtx.E2EConfig.GetVariable(shared.AwsNodeMachineType), nil)
+			unencryptedUDMachineTemplate.Spec.Template.Spec.Ignition = &infrav1.Ignition{
+				StorageType: infrav1.IgnitionStorageTypeOptionUnencryptedUserData,
+			}
+
+			unencryptedUDMachineDeployment := makeMachineDeployment(namespace.Name, unencryptedMDName, clusterName, nil, int32(1))
+			framework.CreateMachineDeployment(ctx, framework.CreateMachineDeploymentInput{
+				Creator:                 e2eCtx.Environment.BootstrapClusterProxy.GetClient(),
+				MachineDeployment:       unencryptedUDMachineDeployment,
+				BootstrapConfigTemplate: makeJoinBootstrapConfigTemplate(namespace.Name, unencryptedMDName),
+				InfraMachineTemplate:    unencryptedUDMachineTemplate,
+			})
+
+			framework.WaitForMachineDeploymentNodesToExist(ctx, framework.WaitForMachineDeploymentNodesToExistInput{
+				Lister:            e2eCtx.Environment.BootstrapClusterProxy.GetClient(),
+				Cluster:           result.Cluster,
+				MachineDeployment: unencryptedUDMachineDeployment,
+			}, e2eCtx.E2EConfig.GetIntervals("", "wait-unencrypted-userdata-worker-node")...)
+
+			unencryptedUDWorkerMachines := framework.GetMachinesByMachineDeployments(ctx, framework.GetMachinesByMachineDeploymentsInput{
+				Lister:            e2eCtx.Environment.BootstrapClusterProxy.GetClient(),
+				ClusterName:       clusterName,
+				Namespace:         namespace.Name,
+				MachineDeployment: *unencryptedUDMachineDeployment,
+			})
+			Expect(len(unencryptedUDWorkerMachines)).To(Equal(1))
+			// There is only one machine.
+			m := unencryptedUDWorkerMachines[0]
+			mScope := &scope.MachineScope{Machine: &m}
```

Comment:
From CI:

```
./unmanaged_functional_test.go:1104:72: cannot use m (variable of type "sigs.k8s.io/cluster-api/api/v1beta1".Machine) as *"sigs.k8s.io/cluster-api/api/v1beta1".Machine value in struct literal
./unmanaged_functional_test.go:1104:75: cannot call pointer method GetRawBootstrapDataWithFormat on "sigs.k8s.io/cluster-api-provider-aws/v2/pkg/cloud/scope".MachineScope
./unmanaged_functional_test.go:1107:38: cannot use m.Spec.ProviderID (variable of type *string) as string value in argument to assertUnencryptedUserDataIgnition
```

Reply:
Yes 👍  I'm fixing those.
Unfortunately my editor's language server seems to be broken at the moment, and it is not catching/reporting these issues, so CI is my LSP atm 😓

Reply:
Fixed. Thanks